source/Tutorials/Advanced/Security/Deployment-Guidelines.rst

```diff
@@ -0,0 +1,214 @@
+Let's start by creating a workspace tree with the following sub-folders:
+
+.. code-block:: bash
+
+  mkdir -p ~/deploy_gd_tutorial/organization_system
+  mkdir ~/deploy_gd_tutorial/remote_system
+  mkdir ~/deploy_gd_tutorial/keystore
+
+Similarly to previous tutorials, intialize a new keystore tree directory.
+This will create enclaves/ public/ and private/ directories, which are explained in more detail in `ROS2 Security enclaves <https://design.ros2.org/articles/ros2_security_enclaves.html>`_.
+
+.. code-block:: bash
+
+  # Source ROS installation
+  source /opt/ros/${ROS_DISTRO}/setup.bash
+  # Initialize a new keystore directory
+  ros2 security create_keystore ~/deploy_gd_tutorial/keystore
+
+At this point, step into the remote_system workspace, create the corresponding enclave and copy just the public/ and enclaves/ directories to the current one.
+Those security artifacts will be needed by the remote system to enable listener's security.
+For the sake of simplicity, the same CA is used within this enclave for both, identity and permissions.
+Note that private/ folder is not moved but left in localhost (organization).
+
+.. code-block:: bash
+
+  # Move to remote's system path
+  cd ~/deploy_gd_tutorial/remote_system
+  # Create an enclave for the secure listener
+  ros2 security create_enclave ~/deploy_gd_tutorial/keystore /talker_listener/listener
+  # Ship enclaves/ and public/ directories only, minimizing security threat
+  cp -R ../keystore/enclaves ../keystore/public .
+
+For the purpose of running a secure listener at the docker image startup, a new ``entrypoint.sh`` file is required with the following content:
+
+.. code-block:: bash
+
+  #!/bin/bash
+  source /opt/ros/${ROS_DISTRO}/setup.bash
+  ros2 run demo_nodes_cpp listener --ros-args --enclave /talker_listener/listener $@
+
+In order to build a new docker image, a Dockerfile is also needed within this directory.
+Create a new file ``Dockerfile`` in the same directory with preferred text editor.
+
+.. code-block:: bash
+
+  ARG ROS_DISTRO=humble
+  FROM ros:${ROS_DISTRO}-ros-base
+
+  RUN apt-get update && apt-get install -y \
+        ros-${ROS_DISTRO}-demo-nodes-cpp \
+        ros-${ROS_DISTRO}-demo-nodes-py && \
+      rm -rf /var/lib/apt/lists/*
+
+  ARG KEYSTORE_DIR=/keystore
+
+  RUN mkdir -p ${KEYSTORE_DIR}/enclaves \
+    mkdir ${KEYSTORE_DIR}/public
+
+  COPY enclaves ${KEYSTORE_DIR}/enclaves
+  COPY public ${KEYSTORE_DIR}/public
+
+  ENV ROS_SECURITY_KEYSTORE=${KEYSTORE_DIR}
+  ENV ROS_SECURITY_ENABLE=true
+  ENV ROS_SECURITY_STRATEGY=Enforce
+
+  COPY entrypoint.sh /entrypoint.sh
+
+  ENTRYPOINT ["/entrypoint.sh"]
+
+Build the docker image with the command:
+
+.. code-block:: bash
+
+  # Build remote's system image
+  docker build -t ros2_security/deployment_tutorial .
+
+Finally, create an enclave for the local talker node.
+
+.. code-block:: bash
+
+  # Go to organization_system path
```

Comment:
My first `docker build` failed because of my I had not created the files where they were supposed to go, but I continued with the rest of the tutorial anyways. These meant that I created the talker enclave. Then, I fix the docker build and ended up with both listener and talker enclaves in the container (yikes!).

Reply:
I created a subsection for the targets device docker image and moved this right after keystore's creation